src/main/java/org/jenkinsci/plugins/badge/PublicBadgeAction.java

```diff
@@ -127,11 +128,11 @@ public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @Que
         Job<?, ?> project = getProject(job);
         Integer codeCoverage = null;
 
-        if (project.getLastSuccessfulBuild() != null) {
```

Comment:
Well I don't know if the latest failed build accurately reflects the code coverage of the repo. Its the latest successful build that does.